Add sensei_duplicate_post_args filter

alexsanford opened the pull request:
Fixes #4165

### Changes proposed in this Pull Request

- Add `sensei_duplicate_post_args` filter for customizing the arguments to `wp_insert_post` when duplicating a Sensei post. This may be a Course, Lesson, or Quiz.
- Remove default `(Duplicate)` suffix from the title of duplicated posts.

_Note that if the `3.10.1` milestone is merged into the `3.11` milestone, we will need to update the doc on this new hook._

### Testing instructions

- Add the following snippet:

```php
add_filter( 'sensei_duplicate_post_args', function( $new_post, $post ) {
	if ( ! in_array( get_post_type( $post ), [ 'lesson', 'course' ], true ) ) {
		return $new_post;
	}

	$new_post['post_status'] = 'publish';
	
	return $new_post;
}, 10, 2 );
```

- Duplicate a Course using the "Duplicate (with lessons)" link on the "All courses" page.
- Ensure that the duplicated Course and all of the duplicated Lessons are published (not draft) and the titles are not suffixed with the `(Duplicate)` text.

<!-- Add the following only if there are new/updated actions or filters. Please provide a brief description of what they do and any arguments they may take. Be sure to also add the "Hooks" label to this PR. -->
### New/Updated Hooks

- `sensei_duplicate_post_args`: Filter arguments for `wp_insert_post` when duplicating a Sensei post. This may be a Course, Lesson, or Quiz.

jom reviewed (approved):
Looks good!

donnapep commented:
For the title, I wonder if we should also remove the word "Duplicate"? I recall other users disliking this naming convention as well for the same reason reported in the issue.

alexsanford commented:
> For the title, I wonder if we should also remove the word "Duplicate"? I recall other users disliking this naming convention as well for the same reason reported in the issue.

Yeah, I think that makes a lot of sense. Changed in https://github.com/Automattic/sensei/pull/4198/commits/57f53ff618e060b44b216b9bdc64a767a5e8f1c6 and I've updated the PR description and snippet for testing.
